src/wasm-linker.cpp

```diff
@@ -210,6 +210,15 @@ void Linker::layout() {
     auto* func = out.wasm.getFunction(name);
     func->type = ensureFunctionType(getSig(func), &out.wasm)->name;
   }
+
+  // Export malloc and free whenever availble. JavsScript version of malloc has
```

Comment:
Yes, good point, it actually isn't the issue. This does send us in the right direction, though - `emcc.py` adds those methods to `EXPORTED_FUNCTIONS`, and that is used throughout emscripten (e.g. when running LLVM opts it makes sure to keep them alive) as well as exporting them from the asm.js module. So what we need is for emscripten to forward `EXPORTED_FUNCTIONS` to `s2wasm` (as a new parameter to `s2wasm`, sent from `emscripten.py`), and for it to export them from the wasm module, to parallel the asm.js case.